src/nvim/os/env.c

```diff
@@ -423,43 +423,41 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(const char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
-  int vimruntime;
-
-
-  p = (char_u *)os_getenv((char *)name);
-  if (p != NULL && *p == NUL)       /* empty is the same as not set */
-    p = NULL;
-
-  if (p != NULL) {
-    return p;
+  const char *kos_env_path = os_getenv(name);
+  if (kos_env_path != NULL
+      && *kos_env_path == NUL) /* empty is the same as not set */
+    kos_env_path = NULL;
+
+  if (kos_env_path != NULL) {
+    *mustfree = true;
+    return xstrdup(kos_env_path);
```

Comment:
> @elmart, was this your suggestion?

Yes, I suggested trying to refactor to const where possible, and using `xstrdup` instead of casting to `char *` when needing to transform to non-const to avoid excessive change cascading.

I don't see any problem with this change. Previously, it returned a const (incorrectly casted to non-const) with mustfree = false; Now, it returns an allocated string with mustfree = true;